incubator/catalog-schema-registry/src/main/java/io/aklivity/zilla/runtime/catalog/schema/registry/internal/SchemaRegistryCatalogHandler.java

```diff
@@ -22,32 +21,49 @@
 import java.text.MessageFormat;
 
 import io.aklivity.zilla.runtime.catalog.schema.registry.internal.config.SchemaRegistryCatalogConfig;
+import io.aklivity.zilla.runtime.catalog.schema.registry.internal.serializer.RegisterSchemaRequest;
 import io.aklivity.zilla.runtime.engine.catalog.CatalogHandler;
-import io.aklivity.zilla.runtime.engine.config.CatalogConfig;
 
 public class SchemaRegistryCatalogHandler implements CatalogHandler
 {
     private static final String SUBJECT_VERSION_PATH = "/subjects/{0}/versions/{1}/schema";
     private static final String SCHEMA_PATH = "/schemas/ids/{0}/schema";
+    private static final String REGISTER_SCHEMA_PATH = "/subjects/{0}/versions";
 
-    private final SchemaRegistryCatalogConfig config;
     private final HttpClient client;
     private final String baseUrl;
+    private final RegisterSchemaRequest request;
 
     public SchemaRegistryCatalogHandler(
-        CatalogConfig catalog)
+        SchemaRegistryCatalogConfig config)
     {
-        this.config = SchemaRegistryCatalogConfig.class.cast(catalog.options);
         this.baseUrl = config.url;
         this.client = HttpClient.newHttpClient();
+        this.request = new RegisterSchemaRequest();
     }
 
     @Override
     public int register(
-        String schema,
-        String subject)
+        String subject,
+        String type,
+        String schema)
     {
-        return 0;
+        int schemaId = 0;
+        HttpRequest httpRequest = HttpRequest
+            .newBuilder(toURI(baseUrl, MessageFormat.format(REGISTER_SCHEMA_PATH, subject)))
+            .header("Content-Type", "application/json")
+            .POST(HttpRequest.BodyPublishers.ofString(request.buildRequestBody(type, schema)))
+            .build();
+        try
+        {
+            HttpResponse<String> response = client.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+            schemaId = response.statusCode() == 200 ? request.resolveRegisterResponse(response.body()) : 0;
```

Comment:
Is `0` a valid value that can be returned from schema registry?

Are we saying that we'll use a special value to indicate registration error? If so, we should define it as a constant on the interface so that callers can use the constant to detect registration error.

Something like `NO_SCHEMA_ID`.